cmd/livepeer/livepeer.go

```diff
@@ -204,19 +183,277 @@ func main() {
+	cfg.maxPricePerUnit = flag.Int("maxPricePerUnit", *cfg.maxPricePerUnit, "The maximum transcoding price (in wei) per 'pixelsPerUnit' a broadcaster is willing to accept. If not set explicitly, broadcaster is willing to accept ANY price")
+	// Unit of pixels for both O's basePriceInfo and B's MaxBroadcastPrice
+	cfg.pixelsPerUnit = flag.Int("pixelsPerUnit", *cfg.pixelsPerUnit, "Amount of pixels per unit. Set to '> 1' to have smaller price granularity than 1 wei / pixel")
+	cfg.autoAdjustPrice = flag.Bool("autoAdjustPrice", *cfg.autoAdjustPrice, "Enable/disable automatic price adjustments based on the overhead for redeeming tickets")
+	// Interval to poll for blocks
+	cfg.blockPollingInterval = flag.Int("blockPollingInterval", *cfg.blockPollingInterval, "Interval in seconds at which different blockchain event services poll for blocks")
+	// Redemption service
+	cfg.redeemer = flag.Bool("redeemer", *cfg.redeemer, "Set to true to run a ticket redemption service")
+	cfg.redeemerAddr = flag.String("redeemerAddr", *cfg.redeemerAddr, "URL of the ticket redemption service to use")
+	// Reward service
+	cfg.reward = flag.Bool("reward", false, "Set to true to run a reward service")
+	// Metrics & logging:
+	cfg.monitor = flag.Bool("monitor", *cfg.monitor, "Set to true to send performance metrics")
+	cfg.metricsPerStream = flag.Bool("metricsPerStream", *cfg.metricsPerStream, "Set to true to group performance metrics per stream")
+	cfg.metricsExposeClientIP = flag.Bool("metricsClientIP", *cfg.metricsExposeClientIP, "Set to true to expose client's IP in metrics")
+	cfg.metadataQueueUri = flag.String("metadataQueueUri", *cfg.metadataQueueUri, "URI for message broker to send operation metadata")
+	cfg.metadataAmqpExchange = flag.String("metadataAmqpExchange", *cfg.metadataAmqpExchange, "Name of AMQP exchange to send operation metadata")
+	cfg.metadataPublishTimeout = flag.Duration("metadataPublishTimeout", *cfg.metadataPublishTimeout, "Max time to wait in background for publishing operation metadata events")
+
+	// Storage:
+	cfg.datadir = flag.String("datadir", *cfg.datadir, "Directory that data is stored in")
+	cfg.objectstore = flag.String("objectStore", *cfg.objectstore, "url of primary object store")
+	cfg.recordstore = flag.String("recordStore", *cfg.recordstore, "url of object store for recordings")
+
+	// API
+	cfg.authWebhookURL = flag.String("authWebhookUrl", *cfg.authWebhookURL, "RTMP authentication webhook URL")
+	cfg.orchWebhookURL = flag.String("orchWebhookUrl", *cfg.orchWebhookURL, "Orchestrator discovery callback URL")
+	cfg.detectionWebhookURL = flag.String("detectionWebhookUrl", *cfg.detectionWebhookURL, "(Experimental) Detection results callback URL")
+
+	return cfg
+}
+
+// updateNilsForUnsetFlags changes some cfg fields to nil if they were not explicitly set with flags.
+// For some flags, the behavior is different whether the value is default or not set by the user at all.
+func updateNilsForUnsetFlags(cfg LivepeerConfig) LivepeerConfig {
```

Comment:
Looks good, I've always found it annoying that the Go flags package doesn't give a way to distinguish between zero value / not set